optimum/neuron/pipelines/transformers/base.py

```diff
@@ -0,0 +1,238 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Pipelines running different backends."""
+
+import logging
+from typing import Any, Dict, Optional, Union
+
+from transformers import (
+    AutoConfig,
+    FeatureExtractionPipeline,
+    FillMaskPipeline,
+    Pipeline,
+    PreTrainedTokenizer,
+    PreTrainedTokenizerFast,
+    QuestionAnsweringPipeline,
+    SequenceFeatureExtractor,
+    TextClassificationPipeline,
+    TokenClassificationPipeline,
+)
+from transformers import pipeline as transformers_pipeline
+from transformers.feature_extraction_utils import PreTrainedFeatureExtractor
+from transformers.onnx.utils import get_preprocessor
+
+from optimum.neuron.modeling_base import NeuronBaseModel
+
+from ...modeling import (
+    NeuronModelForFeatureExtraction,
+    NeuronModelForMaskedLM,
+    NeuronModelForQuestionAnswering,
+    NeuronModelForSequenceClassification,
+    NeuronModelForTokenClassification,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+NEURONX_SUPPORTED_TASKS = {
+    "feature-extraction": {
+        "impl": FeatureExtractionPipeline,
+        "class": (NeuronModelForFeatureExtraction,),
+        "default": "distilbert-base-cased",
+        "type": "text",  # feature extraction is only supported for text at the moment
+    },
+    "fill-mask": {
+        "impl": FillMaskPipeline,
+        "class": (NeuronModelForMaskedLM,),
+        "default": "bert-base-cased",
+        "type": "text",
+    },
+    "question-answering": {
+        "impl": QuestionAnsweringPipeline,
+        "class": (NeuronModelForQuestionAnswering,),
+        "default": "distilbert-base-cased-distilled-squad",
+        "type": "text",
+    },
+    "text-classification": {
+        "impl": TextClassificationPipeline,
+        "class": (NeuronModelForSequenceClassification,),
+        "default": "distilbert-base-uncased-finetuned-sst-2-english",
+        "type": "text",
+    },
+    "token-classification": {
+        "impl": TokenClassificationPipeline,
+        "class": (NeuronModelForTokenClassification,),
+        "default": "dbmdz/bert-large-cased-finetuned-conll03-english",
+        "type": "text",
+    },
+}
+
+
+def load_pipeline(
+    model,
+    targeted_task,
+    load_tokenizer,
+    tokenizer,
+    feature_extractor,
+    load_feature_extractor,
+    SUPPORTED_TASKS,
```

Comment:
```suggestion
    supported_tasks,
```
Is this a global constant or meant to be a function parameter?